editor/filesystem_dock.cpp

```diff
@@ -1695,21 +1695,33 @@ String FileSystemDock::_get_unique_name(const FileOrFolder &p_entry, const Strin
 	return new_path;
 }
 
-void FileSystemDock::_update_favorites_list_after_move(const HashMap<String, String> &p_files_renames, const HashMap<String, String> &p_folders_renames) const {
-	Vector<String> favorites_list = EditorSettings::get_singleton()->get_favorites();
-	Vector<String> new_favorites;
-
-	for (const String &old_path : favorites_list) {
+void FileSystemDock::_update_favorites_after_move(const HashMap<String, String> &p_files_renames, const HashMap<String, String> &p_folders_renames) const {
+	Vector<String> favorite_files = EditorSettings::get_singleton()->get_favorites();
+	Vector<String> new_favorite_files;
+	for (const String &old_path : favorite_files) {
 		if (p_folders_renames.has(old_path)) {
-			new_favorites.push_back(p_folders_renames[old_path]);
+			new_favorite_files.push_back(p_folders_renames[old_path]);
 		} else if (p_files_renames.has(old_path)) {
-			new_favorites.push_back(p_files_renames[old_path]);
+			new_favorite_files.push_back(p_files_renames[old_path]);
 		} else {
-			new_favorites.push_back(old_path);
+			new_favorite_files.push_back(old_path);
 		}
 	}
+	EditorSettings::get_singleton()->set_favorites(new_favorite_files);
+
+	HashMap<StringName, PackedStringArray> favorite_properties = EditorSettings::get_singleton()->get_favorite_properties();
+	for (HashMap<StringName, PackedStringArray>::Iterator I = favorite_properties.begin(); I;) {
+		HashMap<StringName, PackedStringArray>::Iterator next = I;
+		++next;
 
-	EditorSettings::get_singleton()->set_favorites(new_favorites);
+		String old_path = I->key;
+		if (p_files_renames.has(old_path)) {
+			favorite_properties.replace_key(old_path, p_files_renames[old_path]);
```

Comment:
Wouldn't it be simpler to instead iterate on renames and replace favorites if they match?